### ISSUE TITLE
Should we restrict the tag-input for "[" and "]" ?

### ISSUE THREAD

**Issue by BurningTreeC** [Jermolene/TiddlyWiki5#3443]
Hi devs, I've recently made some PR's which work fine with all kinds of tag-names

The only ones that make problems are tags with `[` and `]` in their names - for obvious reasons

I'd like to ask if it wouldn't be the best solution to restrict the tags-input so that it doesn't add tags that contain `[` or `]` ?

What do you think?

all the best, Simon

**Comment by Jermolene** [comment]
Hi @BurningTreeC I'd been hoping that we might make the core resilient to tags containing square brackets. It precludes us from using `[tag[$tag$]]` but we've got some reasonable workarounds now. What do you think? (I'm reading emails backwards so you may already have addressed this point).

**Comment by BurningTreeC** [comment]
@Jermolene yes I'm getting comfortable with the `<__tag__>` syntax now. Making the core resilient to tags containing square brackets would be the last missing puzzle part. I have no idea how but I'd love to have it

**Comment by pmario** [comment]
hmm, I never had that problem and I can't remember many group discussion. So I have no strong opinion on this one.

**Comment by AnthonyMuscio** [comment]
The only reason I ever thought I needed [[ ]] in a title or in fact ~ was when listing in the circumstance below.

My argument is is the issue of tags that contain [ or ] a second order issue generated by another somewhere?. Personally it seems insane to use something that **delimits a title, in a title**, be it a tag or otherwise.

Am I missing something that I seem to need to use the following when listing tiddlers, that themselves contain a camel case words to get links that operate correctly where ever I click on them?

`<$link to=<<tiddlername>> tooltip="Custom tooltip"><$view tiddler=<<tiddlername>> format=text /></$link>`

If this is the only way, then I imagine others may be using `[[ ]]` in an attempt to resolve this or a related issue.

Regrads
Tony

**Comment by BurningTreeC** [comment]
@AnthonyMuscio 

this issue arises from an attempt to harden the tag-pills, the Tag Manager and the `list-tagged-draggable` macro, see #3437, #3440  and #3441 

TiddlyWiki allows titles that contain all kinds of characters, although it shows a warning

If we create a tiddler that contains `]]` (even if it's unlikely that users do) and we create tiddlers tagged with that one, the tags will break. If not instantly, then after reloading the wiki.

**Comment by BurningTreeC** [comment]
This problem then causes a series of other related problems in the Tag Manager and the list-tagged-draggable macro

**Comment by pmario** [comment]
There is a warning, if users create titles, that contain "special" characters. We could enforce this rule and remove those characters from titles. ... BUT this would be an "offence" for me. ... 

For some users "special" chars may work just fine, for their usecase and usage pattern with TW. They just can't use advanced features. 

I'm fine if we can "harden" the core, without increasing the complexity! 

If we increase the complexity for "hypthetical" problems. I'm not a big fan!

just my thoughts.

**Comment by BurningTreeC** [comment]
@pmario 

I share your opinion about this. This comes from trying many different characters for tag names. First I encountered the tags-dropdown in the tags-input field break with quotes. That should be solved with #3437 

Then I found `!!` breaking the dropdown buttons in the Tag Manager beause of the reveal and button widgets interpreting them as `State Reference` - therefor @Jermolene proposed "stateTitle", "stateField" and "stateIndex" attributes for widgets. These attributes would interpret "stateTitle" as a Title, no matter if there are state-reference-like characters.

I also found problems with characters that break filters in the `list-tagged-draggable` macro which should be solved with #3442 

After these findings I've seen that `]]` still breaks tags because of how they're stored, using the `[[]]` syntax. That's why I made this issue allowing us to discuss this
